src/de/grobox/liberario/Preferences.java

```diff
@@ -98,6 +105,44 @@ public static void setPref(Context context, String pref, boolean value) {
 		editor.commit();
 	}
 
+	public static EnumSet<Product> getProducts(Context cont) {
+		// check if there's a preference for selected products for the current network
+		TransportNetwork current_network = Preferences.getTransportNetwork(cont);
+		String network_pref = current_network.getId().toString() + "_selected_products";
+		SharedPreferences sp = cont.getSharedPreferences(Preferences.NETWORK_SETTINGS, Context.MODE_PRIVATE);
```

Comment:
So you are just using one preferences file to store preferences for *all* networks. That is fine as well, but in this case, you can just re-use the existing `PREFS` constant and you don't need your own. Also no need anymore to prefix with `Preferences.`.